C++/2. Variables.md

```diff
@@ -2,59 +2,78 @@
 - `int`
+- `long`
 - `float`
+- `long long`
 - `double`
-- `std::string`
-- `char`
-- `bool`
 
 ## Explanation
 
-The `int` variable type means an integer, which means a whole number. \
-The `float` variable type means a single-precision decimal value. \
-The `double` variable type means a double-precision decimal value. \
-The `std::string` variable type means a string or a / some word(s). Strings are encased in double quotes "". \
-The `char` variable type means a character. `char` stands for character, which means one number or letter. `char`s are encased in single quotes ''. \
-The `bool` variable type means a `true` or `false` variable. The values it can be are those itself.
+The `bool` type means a boolean, which can only be `true` or `false`. The values it can be are those itself.
+The `char` type means a (ASCII) character. `char` stands for character, which means one number or letter. `char`s are encased in single quotes ''.
+The `short` type means a short integer. It can usually hold integers as smaller than `int`, and may take less space.
+The `int` variable type means an integer, which means a whole number.
+The `unsigned` type means an unsigned integer. Unsigned refers to how it lacks a sign, as in a `+` or `-` sign. For example, `4` is a valid value of `unsigned`, whereas `-4` is not. It can hold (slightly) higher positive values, yet no negatives.
+The `long` variable type means a long integer, it can hold up to values as large as or larger than `int`.
+The `float` variable type means a single-precision decimal value.
+The `long long` variable type means a longer integer, it can hold up to values as large as or larger than `long`.
+The `double` variable type means a double-precision decimal value.
+
+**Note: No primitive has infinite precision, for example, `short`s usually cannot hold values <= 70000, and `char` can only hold [ASCII](https://www.ascii-code.com/) or [ASCII-Extended](https://www.ascii-code.com/#extendedASCIIDescription).**
 
 ## Declaration
 
-You declare a variable by typing the variable type before the name of the variable. You then give an equal to sign an write the value you want the variable to contain.
+You declare a variable by typing the variable type before the name of the variable. You then give an equal to sign and write the value you want the variable to contain.
 
 ### Code
 
 ```cpp
 #include <iostream>
 
 int main() {
-  int number = 4;
+  bool boolean = true;
+  char character = 'k';
+  short smallNumber = 2;
+  int number = -4;
+  unsigned positive = 4;
+  long largeNumber = 7000000;
   float decimal = 3.5;
+  long long largerNumber = 70000000;
   double bigDecimal = 3.456;
-  std::string aString = "Hello";
-  char character = 'k';
-  bool boolean = true;
 
-  std::cout << number << std::endl
-      << decimal << std::endl
-      << bigDecimal << std::endl
-      << aString << std::endl
-      << character << std::endl
+  std::cout << std::boolalpha // ignore boolalpha, it will be mentioned later
```

Comment:
is this needed?

Reply:
Well, unless you care to explain to a beginner, who was *just* told that `true` and `false` are the only values that `bool` can hold, why it's outputting `0` and `1`? Really, the 0 and 1 break the abstraction that booleans are intended to create.

Reply:
no no no the boolean only appears once. Why do you have to state that before everything.

Reply:
I'm not sure what you're saying?

Reply:
what I mean is that in the std::cout statement, you only print a Boolean once. This isn't a problem; I was just telling you. You could separate all the normal statements and the one bool alpha statement. First print out the value of the normal statements first, then print out the Boolean in a separate line (I know there aren't 'normal' statements, but it is easier saying that)

Reply:
But... why? What does the separation achieve.

Reply:
O nothing. You don't have to separate it. Do it if you think it would be more clean by doing so